### PR TITLE
PE-27 | Created side menu

### PR DESCRIPTION
**Pull request by ayeejane**
What I did: 

- Created a drawer for the Side Menu.

- Created individual screens for Home, Profile, New Post, and Settings. 

How to Test: 

- Click on the Side Menu bar and see if a drawer pops out with the corresponding texts: 'Home', 'Profile', 'Create New Post', and 'Settings'.

- Then click on each individual text to see if it redirects to a new designated screen and if it goes back to the Home Screen.

### CODE REVIEW

File: lib/main.dart [hunk 1]
```diff
@@ -1,117 +1,60 @@
...
+    BuildContext context, String string, Widget newScreen) {
+  return ListTile(
+      title: Text(string),
+      onTap: () {
+        Navigator.push(
+          context,
+          MaterialPageRoute(builder: (context) => newScreen),
+        );
+      });
+}
```
Comment (tonynguyen98) [inline lib/main.dart]: Please have this in a separate file

**Review by tonynguyen98 — CHANGES_REQUESTED** [review]
Working on my device, please look at my requested changes about extracting the items out of `main.dart`

**Comment by davidsquines** [comment]
Everything else looks good. On Mac, the android emulator loads the app and the sidebar shows. Every single page can be navigated to and the "return to home page" works as intended.

**Review by davidsquines — APPROVED** [review]

File: lib/main.dart [hunk 2]
```diff
@@ -1,117 +1,60 @@
...
+            DrawerHeader(
+              child: Text('Home'),
+              decoration: BoxDecoration(
+                color: Colors.blue,
+              ),
             ),
+            sideMenuListTile(context, 'Home', HomeScreen()),
+            sideMenuListTile(context, 'Profile', ProfileScreen()),
+            sideMenuListTile(context, 'Create New Post', NewPostScreen()),
+            sideMenuListTile(context, 'Settings', SettingsScreen()),
```
Comment (txt020) [inline lib/main.dart]: Looks Good!

**Review by txt020 — APPROVED** [review]

**Comment by ayeejane** [comment]
Created a separate file for side menu inside 'components' folder.

**Review by tonynguyen98 — APPROVED** [review]
Good job! I was able to replicate it on my device and it works very well. Future note, we might need to change `Navigator.push` to `Navigator.pushReplacement`, but don't worry about that for now.

**Pull request merged**